features/nfc/stack/transceiver/pn512/pn512_hw.h

```diff
@@ -0,0 +1,81 @@
+/**
+ * \file pn512_hw.h
+ * \copyright Copyright (c) ARM Ltd 2013
+ * \author Donatien Garnier
+ */
+
+#ifndef PN512_HW_H_
+#define PN512_HW_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "inc/nfc.h"
+
+#include "pn512.h"
+
+//Utility for transport: SPI address read/write
+#define PN512_SPI_ADDR_R(x) ((1<<7) | ((x) << 1))
+#define PN512_SPI_ADDR_W(x) ((0<<7) | ((x) << 1))
+
+void pn512_hw_init(pn512_t* pPN512);
+
+/** \internal Write bytes at the specified address on the underlying transport link
+ * \param pPN512 pointer to pn512_t structure
+ * \param addr address at which to write
+ * \param buf buffer to write
+ * \param len length of buffer
+ */
+static inline void pn512_hw_write(pn512_t* pPN512, uint8_t addr, uint8_t* buf, size_t len)
+{
+  nfc_transport_write(((nfc_transceiver_t*)pPN512)->pTransport, addr, buf, len);
```

Comment:
This type of cast are unnecessary and can lead to unexpected error if the structure gets reordered to reduce padding or a new member gets inserted there.